src/System.Console/src/System/Console.cs

```diff
@@ -25,33 +25,16 @@ public static class Console
         private static ConsoleCancelEventHandler _cancelCallbacks;
         private static ConsolePal.ControlCHandlerRegistrar _registrar;
 
-        internal static T EnsureInitialized<T>(ref T field, Func<T> initializer) where T : class
-        {
-            lock (InternalSyncObject)
-            {
-                T result = Volatile.Read(ref field);
-                if (result == null)
-                {
-                    result = initializer();
-                    Volatile.Write(ref field, result);
-                }
-                return result;
-            }
-        }
+        internal static T EnsureInitializedWrapper<T>(ref T field, Func<T> initializer) where T : class =>
```

Comment:
I renamed it to not cause confusion with LazyInitializer.EnsureInitialized as I wanted to retain this method. In addition, w.r.t. comment below, I wanted to retain this method as it wraps the sync object shared via usages of this EnsureInitialized method. Eliminating this method would require spilling out the lock as well, with potential to use another lock; essentially making the code semantically unequal to what author had written...